packages/ui/src/utils/utilsDate.ts

```diff
@@ -39,3 +39,61 @@ export const convertTimeAgo = (input: Date | number | string) => {
   }
   return `${diffInMinutes}m`
 }
+
+/**
+ * Formats a date into a human-readable string with two variant options.
+ *
+ * @param date - The date to format. Can be a Date object or a string representation of a date.
+ * @param variant - The format variant to use. Defaults to 'short'.
+ *   - 'short': Returns date in format "DD MMM YYYY" (e.g., "31 Jan 2025")
+ *   - 'long': Returns date and time in format "MMM D, YYYY, H:MM AM/PM" (e.g., "Jan 31, 2025, 12:00 AM")
+ *
+ * @returns A formatted date string adjusted for the local timezone.
+ */
+export const formatDate = (date: Date | string, variant: 'short' | 'long' = 'short') => {
+  if (typeof date === 'string') {
+    // `date` is a string when coming from query cache
+    date = new Date(date)
+  }
+  if (variant === 'short') {
+    // example: 31/01/25 adjusted for local timezone
+    return new Intl.DateTimeFormat(undefined, {
+      day: '2-digit',
+      month: 'short',
+      year: 'numeric',
+    }).format(date)
+    // long
+  } else {
+    // example: 31 January, 12:00 AM adjusted for local timezone
+    return new Intl.DateTimeFormat(undefined, {
+      month: 'short',
+      day: 'numeric',
+      year: 'numeric',
+      hour: 'numeric',
+      minute: '2-digit',
+      hour12: true,
+    }).format(date)
+  }
+}
```

Comment:
```suggestion
export const formatDate = (date: Date | string, variant: 'short' | 'long' = 'short') => new Intl.DateTimeFormat(undefined,  variant === 'short' ? {
      // short - example: 31/01/25 adjusted for local timezone
      day: '2-digit',
      month: 'short',
      year: 'numeric',
    } : {
      // long - example: 31 January, 12:00 AM adjusted for local timezone
      month: 'short',
      day: 'numeric',
      year: 'numeric',
      hour: 'numeric',
      minute: '2-digit',
      hour12: true,
    }).format(typeof date === 'string' ? new Date(date) : date)
```

Reply:
alternatively the objects could be in a constant (`Record<Variant, DateOptions>`)